modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-container-test/src/testIntegration/resources/META-INF/resources/runtime_portlet_with_random_instance_id.jsp

```diff
@@ -0,0 +1,24 @@
+<%--
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+--%>
+
+<%@ taglib uri="http://liferay.com/tld/portlet" prefix="liferay-portlet" %>
+
+<%@ page import="com.liferay.portal.kernel.util.StringUtil" %>
+
+<liferay-portlet:runtime
```

Comment:
@dantewang @yuhai I believe we can reuse the runtime_portlet.jsp instead of creating a new one, you just need to setup the request attribute "testRuntimePortletId" to a random string each time it is rendered.

Tina

Reply:
> @dantewang @yuhai I believe we can reuse the runtime_portlet.jsp instead of creating a new one, you just need to setup the request attribute "testRuntimePortletId" to a random string each time it is rendered.

@tinatian, I want to confirm one thing.  In [runtime_portlet.jsp](https://github.com/yuhai/liferay-portal/blob/LPS-83930-pr4/modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-container-test/src/testIntegration/resources/META-INF/resources/runtime_portlet.jsp#L22), it uses ParamUtil.getString(), not uses request.getAttribute(). If we use setAttribute  "testRuntimePortletId" in render(), we need to change ParamUtil.getString() to request.getAttribute() in [runtime_portlet.jsp](https://github.com/yuhai/liferay-portal/blob/LPS-83930-pr4/modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-container-test/src/testIntegration/resources/META-INF/resources/runtime_portlet.jsp#L22). This will affect other cases which used runtime_portlet.jsp (for example, [RenderRequestPortletContainerTest.testIsAccessGrantedByRuntimePortlet](https://github.com/yuhai/liferay-portal/blob/LPS-83930-pr4/modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-container-test/src/testIntegration/java/com/liferay/portal/osgi/web/portlet/container/test/RenderRequestPortletContainerTest.java#L248)). 

Do you mean we also should change other cases (use request.getAttribute() in runtime_portlet.jsp)?

cc @dantewang

Reply:
@tinatian Just an addition to Hai's comment: `ParamUtil.getString()` gets a request parameter. Request parameter can not be modified through request APIs.